packages/webpack/src/utils/module-federation/secondary-entry-points.ts

```diff
@@ -105,6 +105,33 @@ export function recursivelyCollectSecondaryEntryPointsFromDirectory(
   }
 }
 
+function collectPackagesFromExports(
+  pkgName: string,
+  pkgVersion: string,
+  exports: any | undefined,
+  collectedPackages: {
+    name: string;
+    version: string;
+  }[]
+): void {
+  for (const [relativeEntryPoint, exportOptions] of Object.entries(exports)) {
+    if (exportOptions?.['default']?.search(/\.(js|mjs|cjs)$/)) {
```

Comment:
@Coly010 This causes the following error in some cases because for whatever reason search is not a function and it does not handle that gracefully.

```
 NX   exportOptions?.default?.search is not a function
```

Can we update this to the following to have a bit more safety around this and prevent that error?

```suggestion
    if (exportOptions?.['default']?.search?.(/\.(js|mjs|cjs)$/)) {
```